wfe/wfe.go

```diff
@@ -280,6 +285,33 @@ func (wfe *WebFrontEndImpl) HandleFunc(
 	mux.Handle(pattern, defaultHandler)
 }
 
+// processCORS reads and writes all necessary request and response headers in order to
+// enable use by CORS-aware user agents. If the request is a CORS preflight request, the
+// function returns true, in which case no further data may be written to the response.
+func (wfe *WebFrontEndImpl) processCORS(request *http.Request, response http.ResponseWriter,
+	allowedMethods []string) bool {
+	// No Origin header means CORS is not relevant
+	if request.Header.Get("Origin") == "" {
+		return false
+	}
+
+	// All CORS-aware responses include -Allow-Origin and -Expose-Headers
+	response.Header().Set("Access-Control-Allow-Origin", "*")
+	response.Header().Set("Access-Control-Expose-Headers", "Link, Replay-Nonce, Location")
+
+	// Preflight responses additionally include -Max-Age, -Allow-Methods, -Allow-Headers
+	// and terminate the response.
+	if request.Method == http.MethodOptions {
```

Comment:
It's only a valid preflight request if `Access-Control-Request-Method` is set in the request, and the value is a member of allowedMethods (in boulder we assume `GET` if the header is not set, which is probably fine to do here as well).

Reply:
You're right. I've tried to follow the spec a bit more closely on this attempt. I think there's some value in not copying Boulder exactly, so thanks for your review and your patience.